src/include/types.h

```diff
@@ -95,38 +95,47 @@ typedef off_t loff_t;
 // the case of containers of containers. I'm tempted to abstract this
 // stuff using template templates like I did for denc.
 
+namespace std {
```

Comment:
putting these in `namespace std` is technically undefined behavior according to [Extending the std namespace](https://en.cppreference.com/w/cpp/language/extending_std)